### DISCUSSION TITLE
FAN_SOFT_PWN error when compiling

### DISCUSSION THREAD

**Discussion by Ukrevealed** [Jyers/Marlin#2014] [General Discussion]
I have downloaded the most up-to-date firmware and moved the e3v2 no probe config.h and config adv.h to the main marlin folder. 

the only part I have edited in the config.h file is the hotend temp from 275 to 315 as I have installed an all-metal hotend. but when I compile I get at the beginning in yellow

Warning! Ignore unknown configuration option `monitor_flags` in section [env]
Warning! `src_filter` configuration option in section [env:include_tree] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:DUE] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:esp32] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_build_flags` configuration option in section [env:linux_native] is deprecated and will be removed in the next release! Please use `build_src_flags` instead
Warning! `src_filter` configuration option in section [env:linux_native] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_build_flags` configuration option in section [env:simulator_windows] is deprecated and will be removed in the next release! Please use `build_src_flags` instead
Warning! `src_filter` configuration option in section [env:SAMD51_grandcentral_m4] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:malyan_M300] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:STM32F103CB_malyan] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:teensy31] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:teensy35] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:teensy36] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Warning! `src_filter` configuration option in section [env:teensy41] is deprecated and will be removed in the next release! Please use `build_src_filter` instead
Processing STM32F103RE_creality_maple (board: genericSTM32F103RE; platform: ststm32@~12.1; framework: arduino)

And in RED at the end 

In file included from Marlin\src\HAL\STM32F1\../../inc/MarlinConfig.h:49:0,
                 from Marlin\src\HAL\STM32F1\HAL.cpp:30:
Marlin\src\HAL\STM32F1\../../inc/SanityCheck.h:2019:6: error: #error "FAN_SOFT_PWM is required for your board. Enable it to continue."        
     #error "FAN_SOFT_PWM is required for your board. Enable it to continue."
      ^~~~~
In file included from Marlin\src\HAL\STM32F1\../../inc/MarlinConfig.h:49:0,
                 from Marlin\src\HAL\STM32F1\HAL_SPI.cpp:32:
Marlin\src\HAL\STM32F1\../../inc/SanityCheck.h:2019:6: error: #error "FAN_SOFT_PWM is required for your board. Enable it to continue."        
     #error "FAN_SOFT_PWM is required for your board. Enable it to continue."
      ^~~~~
In file included from Marlin\src\HAL\STM32F1\../../inc/MarlinConfig.h:49:0,
                 from Marlin\src\HAL\STM32F1\SPI.cpp:43:
Marlin\src\HAL\STM32F1\../../inc/SanityCheck.h:2019:6: error: #error "FAN_SOFT_PWM is required for your board. Enable it to continue."        
     #error "FAN_SOFT_PWM is required for your board. Enable it to continue."
      ^~~~~
In file included from Marlin\src\HAL\STM32F1\../../inc/MarlinConfig.h:49:0,
                 from Marlin\src\HAL\STM32F1\Servo.cpp:25:
Marlin\src\HAL\STM32F1\../../inc/SanityCheck.h:2019:6: error: #error "FAN_SOFT_PWM is required for your board. Enable it to continue."        
     #error "FAN_SOFT_PWM is required for your board. Enable it to continue."
      ^~~~~
In file included from Marlin\src\HAL\STM32F1\../../inc/MarlinConfig.h:49:0,
                 from Marlin\src\HAL\STM32F1\MarlinSerial.cpp:25:
Marlin\src\HAL\STM32F1\../../inc/SanityCheck.h:2019:6: error: #error "FAN_SOFT_PWM is required for your board. Enable it to continue."        
     #error "FAN_SOFT_PWM is required for your board. Enable it to continue."
      ^~~~~
In file included from Marlin\src\HAL\STM32F1\dogm\../../../inc/MarlinConfig.h:49:0,
                 from Marlin\src\HAL\STM32F1\dogm\u8g_com_stm32duino_swspi.cpp:25:
Marlin\src\HAL\STM32F1\dogm\../../../inc/SanityCheck.h:2019:6: error: #error "FAN_SOFT_PWM is required for your board. Enable it to continue."
     #error "FAN_SOFT_PWM is required for your board. Enable it to continue."
      ^~~~~
In file included from Marlin\src\HAL\STM32F1\../../inc/MarlinConfig.h:49:0,
                 from Marlin\src\HAL\STM32F1\eeprom_bl24cxx.cpp:29:
Marlin\src\HAL\STM32F1\../../inc/SanityCheck.h:2019:6: error: #error "FAN_SOFT_PWM is required for your board. Enable it to continue."        
     #error "FAN_SOFT_PWM is required for your board. Enable it to continue."
      ^~~~~
*** [.pio\build\STM32F103RE_creality_maple\src\src\HAL\STM32F1\HAL.cpp.o] Error 1
In file included from Marlin\src\HAL\STM32F1\HAL_SPI.cpp:33:0:
Marlin\src\HAL\STM32F1/SPI.h:420:13: warning: 'void waitSpiTxEnd(spi_dev*)' defined but not used [-Wunused-function]
 static void waitSpiTxEnd(spi_dev *spi_d) {
             ^~~~~~~~~~~~
*** [.pio\build\STM32F103RE_creality_maple\src\src\HAL\STM32F1\HAL_SPI.cpp.o] Error 1
*** [.pio\build\STM32F103RE_creality_maple\src\src\HAL\STM32F1\Servo.cpp.o] Error 1
*** [.pio\build\STM32F103RE_creality_maple\src\src\HAL\STM32F1\MarlinSerial.cpp.o] Error 1
*** [.pio\build\STM32F103RE_creality_maple\src\src\HAL\STM32F1\SPI.cpp.o] Error 1
*** [.pio\build\STM32F103RE_creality_maple\src\src\HAL\STM32F1\dogm\u8g_com_stm32duino_swspi.cpp.o] Error 1
*** [.pio\build\STM32F103RE_creality_maple\src\src\HAL\STM32F1\eeprom_bl24cxx.cpp.o] Error 1
In file included from Marlin\src\HAL\STM32F1\../../inc/MarlinConfig.h:49:0,
                 from Marlin\src\HAL\STM32F1\eeprom_flash.cpp:32:
Marlin\src\HAL\STM32F1\../../inc/SanityCheck.h:2019:6: error: #error "FAN_SOFT_PWM is required for your board. Enable it to continue."        
     #error "FAN_SOFT_PWM is required for your board. Enable it to continue."
      ^~~~~
*** [.pio\build\STM32F103RE_creality_maple\src\src\HAL\STM32F1\eeprom_flash.cpp.o] Error 1
======================================================== [FAILED] Took 39.59 seconds ========================================================

Environment                 Status    Duration
--------------------------  --------  ------------
STM32F103RE_creality_maple  FAILED    00:00:39.589
=================================================== 1 failed, 0 succeeded in 00:00:39.589 =================================================== 

 *  The terminal process "C:\Users\ukrev\.platformio\penv\Scripts\platformio.exe 'run'" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it. 

Any help would be great.

**Comment by darco415** [comment]
dont use Maple as env

**Reply by Ukrevealed** [reply]
sorry, what do you mean? I am pretty new to the coding side.

**Comment by darco415** [comment]
Umm your not coding your just cut ,pasting,and editing...anyways change the environment from what you have to:
default_envs = STM32F103RE_creality
this is assuming you have the same chip as I do on a 4.2.7 board

**Comment by Ergonomicmike** [comment]
You didn't say it, but I'll assume that you're compiling firmware for an Ender 3 v2.

The newest code for the Ender 3 v2 has customized Configuration files for the Ender.

https://github.com/LCh-77/Marlin/tree/BLTUBL-LA

I suggest using those newer Configuration.h from it.

In any event, as one hacker to another, the Errors that you saw told you what needed to be changed.

So your Configuration.h should have a line like this:

`#define FAN_SOFT_PWM  // Ender Configs`

In previous, generic configuration files, you need to uncomment that line, by removing the // before it.
